core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -162,7 +162,11 @@ object StoreUtils extends Logging {
       val primaryKey = {
         v match {
           case PRIMARY_KEY => ""
-          case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          //case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          case _ => s"$PRIMARY_KEY ($SHADOW_COLUMN_NAME)" // Removing partition column for
+          // primary key. This has been done as we dont have support for CLOB in primary key in
+          // desrailized form. Once that's fixed we should uncomment the above line and comment
+          // this line
```

Comment:
Yes, hash partitioning does support LOBs/UDTs just fine because it uses a hashCode which is implemented for all types.

> There is no equivalent of CLOB at Spark side.

StringType is the equivalent because it provides unlimited size strings.

> What about user partitions based on (StringType, IntegerType) columns?

As discussed in email chain, we want to keep these for now otherwise it will end up creating a global index that can have a major insert performance hit.